onnxruntime/core/providers/dml/DmlExecutionProvider/src/ExecutionProvider.cpp

```diff
@@ -176,6 +176,15 @@ namespace Dml
             sizeof(featureLevels)
             ));
 
+        D3D12_FEATURE_DATA_D3D12_OPTIONS4 featureOptions = {};
+        ORT_THROW_IF_FAILED(d3d12Device->CheckFeatureSupport(
```

Comment:
`D3D12_FEATURE_DATA_D3D12_OPTIONS4` has been around for a while, but it's _possible_ that the user runs on much older build of Win10 where it's not present. Rather than throwing an exception, it's safer to allow failure in the feature check:

```cpp
if (SUCCEEDED(d3d12Device->CheckFeatureSuport(...))
{
    m_native16BitShaderOpsSupported = featureOptions.Native16BitShaderOpsSupported;
}
```